src/main/java/org/opensearch/knn/index/query/KNNQueryFactory.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.query;
+
+import org.apache.lucene.search.KnnVectorQuery;
+import org.apache.lucene.search.Query;
+import org.opensearch.knn.index.util.KNNEngine;
+
+/**
+ * Creates the Lucene k-NN queries
+ */
+public class KNNQueryFactory {
+
+    /**
+     * Creates a Lucene query for a particular engine.
+     *
+     * @param knnEngine Engine to create the query for
+     * @param indexName Name of the OpenSearch index that is being queried
+     * @param fieldName Name of the field in the OpenSearch index that will be queried
+     * @param vector The query vector to get the nearest neighbors for
+     * @param k the number of nearest neighbors to return
+     * @return Lucene Query
+     */
+    public static Query create(KNNEngine knnEngine, String indexName, String fieldName, float[] vector, int k) {
```

Comment:
Lets add debug logger to notify which query builder got created. Lets add  indexName, fieldName. vector, k to the debug log.

Reply:
Added. I left out vector because that might be too much information if there are a lot of queries.